### PR TITLE
feat(datadog/windows): add a init script for datadog on windows

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/2980
and https://github.com/jenkins-infra/helpdesk/issues/3149

### CODE REVIEW

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 1]
```diff
@@ -454,7 +454,9 @@ profile::jenkinscontroller::jcasc:
           virtualNetworkName: "prod-jenkins-public-prod"
           virtualNetworkResourceGroupName: "prod-jenkins-public-prod"
           subnetName: "ci.j-agents-vm"
-          spot: false
+          spot: true
+          initScript: &BootstrapDatadogScriptWindows
+            - (Get-Content C:\ProgramData\Datadog\datadog.yaml -Raw) -Replace 'api_key:', 'api_key: ENC[PKCS7,MIIBmQYJKoZIhvcNAQcDoIIBijCCAYYCAQAxggEhMIIBHQIBADAFMAACAQEwDQYJKoZIhvcNAQEBBQAEggEAqyumVXNZAJylM6NiC/zKT+FKHqGnSwfUMWBupxumSJWL8GHEYFPB/Irr+sRiCPkRWW88bYg7Mp3Rv3qbxc0Aw/cU3XcQZB8RrAOsODBMjOJpHTvoB5aDtSYqygCjffPgwXK0DKguR2ADtsNkMcK7BOT02uaXgZOwo2S5UKkuDn6DsQBq9m1Ho8SwKw/E/+rwsUow+o5JVvPF9//8BIb53Y+wHYm2tOSe5dHFzBIP/+mJMHMpYiE5/q5LBELM9ZXhtBx2zeagEzbriDdlnDkMZk5AMWYz3+omDIX1MO2Fmg9Ak6O4dcNl4hhGJ5KtmPgLGzuhvfh018gi3NVhMBXv9zBcBgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBBltum5EYpT8EqN+vY0wudsgDBX++BA5bC+DpDxuqoANdKkUZdPyBo2cwxq8W4FfUtKE6ZReUIXxvARdR/Z2c5gYY4=]' | Set-Content C:\ProgramData\Datadog\datadog.yaml
```
Comment (dduportal) [inline hieradata/clients/azure.ci.jenkins.io.yaml]: Switching to draft until the init script is completed (as it only performs 1 task in its current state).

File: hieradata/vagrant/common.yaml [hunk 2]
```diff
@@ -333,6 +333,28 @@ profile::jenkinscontroller::jcasc:
...
+          description: "Windows 2019 test datadog"
+          imageDefinition: jenkins-agent-windows-2019
+          os: "windows"
+          storageAccount: SuperSecretThatShouldBeEncryptedInProduction
+          location: "East US 2"
+          instanceType: Standard_D4s_v3 # 4 vCPUS / 16 Gb
+          architecture: amd64
+          labels:
+            - docker-windows-test-datadog
+          idleTerminationMinutes: 30 # Quick deallocation as Linux is billed per minute
```
Comment (dduportal) [inline hieradata/vagrant/common.yaml]:
```suggestion
```

Remove it here, and also on the `hieradata/clients/azure.ci.jenkins.io.yaml`  as per https://github.com/jenkins-infra/jenkins-infra/pull/2401 (a PR we merged during your night so you were not aware of it)

**Comment by smerle33** [comment]
init script manually withing ci.j.io 
win-201-0a3880
Capture d’écran 2022-09-29 à 15 42 43

**Review by dduportal — DISMISSED** [review]
🚢

**Review by dduportal — APPROVED** [review]

**Pull request merged**